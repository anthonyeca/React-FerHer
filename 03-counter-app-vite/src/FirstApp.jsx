import PropTypes from 'prop-types'

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1>{title}</h1>
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: 'Anthony',
  subTitle: 'No hay subtitulo',
  // title: 'No hay titulo',
}
