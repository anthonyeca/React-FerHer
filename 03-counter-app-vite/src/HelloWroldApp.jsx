// export function App(){
//     return (<h1>Hola Mundo</h1>)
//   }

export const HelloWroldApp = () => {
  return <h1>Hello World App</h1>
}
