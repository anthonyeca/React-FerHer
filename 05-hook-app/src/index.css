body {
  font-weight: bold;
  padding: 20px;
}

button {
  margin-right: 10px;
}
