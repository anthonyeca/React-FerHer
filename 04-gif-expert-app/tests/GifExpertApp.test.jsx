import { render, renderHook, screen } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'

describe('pruebas en <GifExpertApp />', () => {
  test('debemos tener una snapshot', () => {
    const { container } = render(<GifExpertApp />)
    expect(container).toMatchSnapshot()
  })
  test('should first', () => {
    const { result } = renderHook(() => GifExpertApp())
    const { categories } = result.current
    console.log(categories)
  })
})
