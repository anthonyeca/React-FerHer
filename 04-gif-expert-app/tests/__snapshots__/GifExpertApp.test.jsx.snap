// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debemos tener una snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
